### PR TITLE
⬆️Maintenance: update mypy to 1.2.0

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

- [x] upgrades mypy to 1.2.0
- [x] use annotation instead of casting in storage

## Related issue/s
relates to https://github.com/ITISFoundation/osparc-issues/issues/675

### CODE REVIEW

**Review by pcrespov — APPROVED** [review]
thx!

**Review by matusdrobuliak66 — APPROVED** [review]
:+1:

File: services/storage/src/simcore_service_storage/datcore_adapter/datcore_adapter.py [hunk 1]
```diff
@@ -227,7 +227,8 @@ async def get_file_download_presigned_link(
         dict[str, Any],
         await _request(app, api_key, api_secret, "GET", f"/files/{file_id}"),
     )
-    return cast(AnyUrl, parse_obj_as(AnyUrl, file_download_data["link"]))  # mypy
+    url: AnyUrl = parse_obj_as(AnyUrl, file_download_data["link"])
```
Comment (colinRawlings) [inline services/storage/src/simcore_service_storage/datcore_adapter/datcore_adapter.py:230]: Is the `: AnyUrl` annotation needed here? It looks to me like `parse_obj_as` is a `Generic`.

Comment (sanderegg) [reply services/storage/src/simcore_service_storage/datcore_adapter/datcore_adapter.py:230]: @colinRawlings if you have some knowledge about this... mypy does not appreciate the Generic.
pylance is happy, but mypy is not.
Either it is a cast which I would like to, or using the annotation. Do you have other ideas?

Comment (sanderegg) [reply services/storage/src/simcore_service_storage/datcore_adapter/datcore_adapter.py:230]: as a referrence, it seems to be a pydantic issue:
https://github.com/pydantic/pydantic/issues/4556

Comment (colinRawlings) [reply services/storage/src/simcore_service_storage/datcore_adapter/datcore_adapter.py:230]: Interesting, I have only experience with `pylance`. I don't think I agree with the `mypy` team's decision to:

> reject the idea of implementing a feature that allows mypy to trust an imported package's type hints as is

**Pull request merged**